x-pack/metricbeat/module/aws/s3_daily_storage/s3_daily_storage.go

```diff
@@ -0,0 +1,220 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package s3_daily_storage
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/x-pack/metricbeat/module/aws"
+)
+
+var metricsetName = "s3_daily_storage"
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each host defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet(aws.ModuleName, metricsetName, New,
+		mb.DefaultMetricSet(),
+	)
+}
+
+// MetricSet holds any configuration or state information. It must implement
+// the mb.MetricSet interface. And this is best achieved by embedding
+// mb.BaseMetricSet because it implements all of the required mb.MetricSet
+// interface methods except for Fetch.
+type MetricSet struct {
+	*aws.MetricSet
+	logger *logp.Logger
+}
+
+// New creates a new instance of the MetricSet. New is responsible for unpacking
+// any MetricSet specific configuration options if there are any.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	cfgwarn.Beta("The aws s3_daily_storage metricset is beta.")
+	s3Logger := logp.NewLogger(aws.ModuleName)
+
+	moduleConfig := aws.Config{}
+	if err := base.Module().UnpackConfig(&moduleConfig); err != nil {
+		return nil, err
+	}
+
+	if moduleConfig.Period == "" {
+		err := errors.New("period is not set in AWS module config")
+		s3Logger.Error(err)
+	}
+
+	metricSet, err := aws.NewMetricSet(base)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating aws metricset")
+	}
+
+	// Check if period is set to be multiple of 86400s
+	remainder := metricSet.PeriodInSec % 86400
+	if remainder != 0 {
+		err := errors.New("period needs to be set to 86400s (or a multiple of 86400s). " +
+			"To avoid data missing or extra costs, please make sure period is set correctly " +
+			"in config.yml")
+		s3Logger.Info(err)
```

Comment:
Just double checking that error shouldn't be returned here

Reply:
Yeah no error, it's just not recommended if the period is not set to that value. Data collection will continue but might lose data or cause extra money.